dot/node_integration_test.go

```diff
@@ -0,0 +1,329 @@
+	require.NoError(t, err)
+
+	stateRoot := genesisHeader.StateRoot
+	expectedHeader, err := types.NewHeader(common.NewHash([]byte{0}), stateRoot, trie.EmptyHash, big.NewInt(0),
+		types.NewDigest())
+	require.NoError(t, err)
+	require.Equal(t, expectedHeader.Hash(), genesisHeader.Hash())
+}
+
+func TestInitNode_LoadStorageRoot(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+
+	cfg.Core.Roles = types.FullNodeRole
+	cfg.Core.BabeAuthority = false
+	cfg.Core.GrandpaAuthority = false
+	cfg.Init.Genesis = genPath
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genPath)
+	require.NoError(t, err)
+
+	err = InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	ks.Gran.Insert(ed25519Keyring.Alice())
+	sr25519Keyring, _ := keystore.NewSr25519Keyring()
+	ks.Babe.Insert(sr25519Keyring.Alice())
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	require.IsType(t, node, &Node{})
+
+	expected := &trie.Trie{}
+	err = expected.LoadFromMap(gen.GenesisFields().Raw["top"])
+	require.NoError(t, err)
+
+	expectedRoot, err := expected.Hash()
+	require.NoError(t, err)
+
+	mgr := node.Services.Get(&core.Service{})
+
+	coreSrvc, ok := mgr.(*core.Service)
+	require.True(t, ok, "could not find core service")
+	require.NotNil(t, coreSrvc)
+
+	stateRoot, err := coreSrvc.StorageRoot()
+	require.NoError(t, err)
+	require.Equal(t, expectedRoot, stateRoot)
+}
+
+func balanceKey(t *testing.T, publicKey [32]byte) (storageTrieKey []byte) {
+	accKey := append([]byte("balance:"), publicKey[:]...)
+	hash, err := common.Blake2bHash(accKey)
+	require.NoError(t, err)
+	return hash[:]
+}
+
+func TestInitNode_LoadBalances(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+
+	cfg.Core.Roles = types.FullNodeRole
+	cfg.Core.BabeAuthority = false
+	cfg.Core.GrandpaAuthority = false
+	cfg.Init.Genesis = genPath
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	ks.Gran.Insert(ed25519Keyring.Alice())
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	require.IsType(t, node, &Node{})
+
+	mgr := node.Services.Get(&state.Service{})
+
+	stateSrv, ok := mgr.(*state.Service)
+	require.True(t, ok, "could not find core service")
+	require.NotNil(t, stateSrv)
+
+	kr, _ := keystore.NewSr25519Keyring()
+	alice := kr.Alice().Public().(*sr25519.PublicKey).AsBytes()
+
+	bal, err := stateSrv.Storage.GetStorage(nil, balanceKey(t, alice))
+	require.NoError(t, err)
+
+	const genbal = "0x0000000000000001"
+	expected, err := common.HexToBytes(genbal)
+	require.NoError(t, err)
+	require.Equal(t, expected, bal)
+}
+
+func TestNode_PersistGlobalName_WhenInitialize(t *testing.T) {
+	globalName := RandomNodeName()
+
+	cfg := NewTestConfig(t)
+	cfg.Global.Name = globalName
+	require.NotNil(t, cfg)
```

Comment:
we already accessed it so it's not nil at this point

```suggestion
```

Reply:
Updated.